hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/AddDurableExecutorConfigMessageTask.java

```diff
@@ -44,7 +44,7 @@ protected ClientMessage encodeResponse(Object response) {
     @Override
     protected IdentifiedDataSerializable getConfig() {
         DurableExecutorConfig config = new DurableExecutorConfig(parameters.name, parameters.poolSize,
-                parameters.durability, parameters.capacity);
+                parameters.durability, parameters.capacity, parameters.statisticsEnabled);
```

Comment:
Default value is `true`. But i wonder in what scenario client doesn't send `statisticsEnabled` to server?

Reply:
`isStatisticsEnabledExists` will be `false` for old clients (v4.0 series)(`statisticsEnabled` will have the default value for its type in this case, but I think we shouldn't rely on this fact and consider `statisticsEnabled` as something undefined) and will be `true` otherwise. (>=v4.1). `statisticsEnabled` will only have a valid value when the `isStatisticsEnabledExists` is `true`.

So, if we want to also enable statistics when the v4.0 client adds an executor config, something like `!parameters.isStatisticsEnabledExists || parameters.statisticsEnabled` would be enough. I initially thought that we would disable statistics when the old client adds an executor config. If that is the case, the snippet in my first comment would work.

Reply:
Thanks for the explanation. It should have been fixed it now.